src/components/drawer/confirm-send-drawer.tsx

```diff
@@ -1,121 +1,132 @@
-import React, { useEffect, useState, useCallback } from 'react';
-import { Button, Stack } from '@stacks/ui';
-import { Text, Caption } from '@components/typography';
+import React, { useCallback, useState } from 'react';
+import { Button, Stack, StackProps } from '@stacks/ui';
+import { Caption, Text } from '@components/typography';
 import { Tooltip } from '@components/tooltip';
 import { BaseDrawer, BaseDrawerProps } from './index';
-import { StacksTransaction, broadcastTransaction } from '@stacks/transactions';
-import { useWallet } from '@common/hooks/use-wallet';
+import { StacksTransaction } from '@stacks/transactions';
 import { stacksValue } from '@common/stacks-utils';
-import { useAnalytics } from '@common/hooks/use-analytics';
-import { ScreenPaths } from '@store/onboarding/types';
-import { useRecoilCallback, useRecoilValue } from 'recoil';
-import { selectedAssetStore } from '@store/recoil/asset-search';
-import { stacksNetworkStore } from '@store/recoil/networks';
+import { useHandleSubmitTransaction } from '@common/hooks/use-submit-stx-transaction';
 import { LoadingRectangle } from '@components/loading-rectangle';
-import { internalTransactionStore } from '@store/recoil/transaction';
-import { correctNonceStore } from '@store/recoil/api';
 import { truncateMiddle } from '@stacks/ui-utils';
+import { useLoading } from '@common/hooks/use-loading';
+
+import { useMakeTransferEffect } from '@common/hooks/use-make-stx-transfer';
+import { useRecoilValue } from 'recoil';
+import { selectedAssetStore } from '@store/recoil/asset-search';
+import { getTicker } from '@common/utils';
 
 interface ConfirmSendDrawerProps extends BaseDrawerProps {
   amount: number;
   recipient: string;
 }
 
-export const ConfirmSendDrawer: React.FC<Omit<ConfirmSendDrawerProps, 'title'>> = ({
-  showing,
-  close,
-  amount,
-  recipient,
-}) => {
-  const [tx, setTx] = useState<StacksTransaction | undefined>();
-  const { doSetLatestNonce } = useWallet();
-  const [loading, setLoading] = useState(false);
+const LOADING_KEY = 'confirm-send-drawer';
+
+const TransactionDetails: React.FC<
+  {
+    amount: number;
+    nonce?: number;
+    fee?: number;
+    recipient: string;
+  } & StackProps
+> = ({ amount, nonce, fee, recipient, ...rest }) => {
   const asset = useRecoilValue(selectedAssetStore);
-  const stacksNetwork = useRecoilValue(stacksNetworkStore);
-  const { doChangeScreen } = useAnalytics();
+  const ticker = asset && asset.type !== 'stx' ? getTicker(asset.name).toUpperCase() : 'STX';
```

Comment:
`isStx` might be a nice helper

Reply:
Another example for `getTicker` to return full caps